### PR TITLE
Milestone 1: Webpack, Json data and auth

### PR DESCRIPTION
**Pull request by willkotheimer**
As, the title says, this lays the foundation for the project. It provides basic auth with firebase and json for the seed data.

## Description

Changes:
Adds 6 json files made from postman for seed data to repopulate firebase if needed
Adds folder structure and implements authentication from firebase
Styles login and logout page and provides functions for same

## Related Issue

[https://github.com/nss-evening-cohort-13/star-destroyer-dashboard-star-destroyer-dashboard/issues/35](https://github.com/nss-evening-cohort-13/star-destroyer-dashboard-star-destroyer-dashboard/issues/35)

## Motivation and Context

This is to provide the foundation for further development

## How Can This Be Tested?

git fetch origin pull/46/head:development

### URL

[https://star-destroyer-a0766.web.app](https://star-destroyer-a0766.web.app)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by brownabs** [comment]
Where is the link for the deployed site?

File: src/javascripts/components/auth/auth.js
Comment (brownabs) [inline src/javascripts/components/auth/auth.js:14]: *Google is misspelled

**Review by drteresavasquez — CHANGES_REQUESTED** [review]
Needs a deploy link in the PR.

**Comment by jacobpaine** [comment]
Looks good. Deploy it on Netlify real quick, so I can check it.

**Comment by drteresavasquez** [comment]
> Looks good. Deploy it on Netlify real quick, so I can check it.

Let’s deploy using Firebase.

**Comment by brownabs** [comment]
Before this is approved, I'd like you all to create the instructions.md file again. It needs to be in your project so that I can continue to refer back to it. Also, I would like for you to explain what your thought process was on having multiple json files in your project. Thanks

**Comment by willkotheimer** [comment]
Alright, I'm npm installing firebase for the project. 

Should have a link in a few!

**Review by brownabs — CHANGES_REQUESTED** [review]

**Comment by willkotheimer** [comment]
Updated with URL!

File: src/javascripts/components/auth/auth.js
Comment (drteresavasquez) [reply src/javascripts/components/auth/auth.js:14]: Looks like they used an icon for the G

File: src/javascripts/helpers/data/authData.js
Comment (drteresavasquez) [inline src/javascripts/helpers/data/authData.js:8]: You Probably did this to get auth working, but this is going to be hard to maintain in the long term if you’re hiding and showing classes for Dom elements

**Review by drteresavasquez — APPROVED** [review]
I am giving my approval for the code and deployed auth, but @brownabs has the final sign off for the merge into main.

**Comment by drteresavasquez** [comment]
@willkotheimer make sure your team knows you added Firebase so when they pull main, they will need to run npm i to install the new packages.

File: src/javascripts/helpers/data/authData.js
Comment (brownabs) [reply src/javascripts/helpers/data/authData.js:8]: ^100% agree!

**Review by brownabs — APPROVED** [review]

**Comment by brownabs** [comment]
You have my approval

File: src/javascripts/helpers/data/authData.js
Comment (wppattonjr) [reply src/javascripts/helpers/data/authData.js:8]: Yeah I just did it to get auth working it will not stay this way. That would be a headache to write lol. Just wanted to clearly show auth works.

**Pull request merged**
